qiskit_nature/circuit/library/__init__.py

```diff
@@ -18,8 +18,14 @@
 
 .. currentmodule:: qiskit_nature.circuit.library
 
+.. autosummary::
+   :toctree: ../stubs/
+   :nosignatures:
+
+   BogoliubovTransform
```

Comment:
Originally I did do that with the title "primitives" (as in algorithmic primitives) but @woodsp-ibm said the name was confusing and suggested simply having it at the top level. Do you have any ideas for the name?

Reply:
Hm I see. I just find it a bit confusing that something called a `Tranform` is apparently a `QuantumCircuit`. How can you reason this? Maybe we can find a better name here?

Reply:
Those initial _states_ are indeed circuits because e.g. the `HartreeFock` case will add `X` gates to a circuit which will place the qubits into `|1>` where we want to initialize a particle (after mapping).

But I see your point about the Fourier transform. Would it even make sense to add a module called `transforms` then?

Reply:
I think `transforms` is both too vague and too specific. It's vague because it's not well-defined, and it's too specific because in practice it will probably only encompass circuits whose common name ends with the word "transform". This is why I prefer the more general term `primitives` for algorithmic primitives. I actually think it's fine to leave it at the top level and worry about creating submodules when we have more examples.

Reply:
Okay in that case let's leave it where it is right now.
Primitives will not be an option due to a name clash in the Qiskit ecosystem which will cause too much confusion.